app/modules/transcribe/transcribe.controller.js

```diff
@@ -93,12 +94,7 @@ function TranscribeController($stateParams, $modal, $scope, $window, Annotations
         vm.subject = SubjectsFactory.current;
     }
 
-    function subjectLoadError(result) {
-        if (result === 'outOfData') {
-            $scope.$broadcast('subject:outOfData');
```

Comment:
Did you mean to delete this?

Reply:
I did, because [your pseudo-code didn't use it](https://github.com/zooniverse/shakespeares_world/pull/369#issuecomment-371157485), so I thought deleting was implied in your comment. Was it not?

Reply:
That code was an example of how to catch errors in a chain of promises, that's all.

What did this line do? It looks, to me, like it notifies when the workflow is out of subjects, or something like that. Won't removing it break SW wherever it listens for this event?

Reply:
You're right. I've been looking into the same 10 lines for too long.

Reply:
Mind you, not adding it back in helps with stopping requests...

Reply:
If I understand correctly, you're adding it back in like this?
```javascript
loadSubjects()
.catch(handleSubjectError)
.then(loadAggregations)
.catch(handleErrors)
```
in which case `loadAggregations` will always be run, even if `loadSubjects` throws an error, because you're catching errors mid-chain then recovering and continuing the chain. Read the section about [Promise chains](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#Chaining) at MDN for more detail.

What you really want to be doing is interrupting the chain, then handling your errors at the end ie. use `handleErrors` to handle any errors that occurred while subjects were loading, or while aggregations were loading.

Reply:
No, I was adding it back like this:
```
function loadSubject() {
        return SubjectsFactory.$getData($stateParams.subjectSet)
            .then(subjectLoaded, subjectLoadError)
            .then(loadAggregations)
            .catch(handleErrors);
    }
```
But I think it's equivalent.

Reply:
Yep, that's functionally the same as the pseudo-code I posted. A `catch` block is equivalent to passing a second argument to `then`. I think the MDN docs explain that too.

Reply:
Should I just remove `subjectLoadError ` then (provided I find another decent place for the `out-of-date` message)?

Reply:
I read that. That's why I said they are equivalent.